tests/integration/test_dss.py

```diff
@@ -36,6 +41,32 @@ def test_status_before_initialize(cleanup_after_initialize) -> None:
     assert "GPU acceleration: Disabled" in result.stdout
 
 
+@pytest.mark.gpu_run
+def test_status_before_initialize_gpu(cleanup_after_initialize) -> None:
```

Comment:
I'm sure there's a way in pytest to have an if statuement in these gpu/cpu tests that could do:
```python
if cpu_test:
  assert "GPU acceleration: Disabled
elif gpu_test:
  ...
```

but I spent a long time and couldn't figure it out :(  We can inspect the marks using `request.node.own_markers`, but that'll show all the markers the test has, not all the markers that match the `pytest -m markers` command.  :(

Reply:
Actually, this does work provided we always specify a mark:

```python
@pytest.mark.parametrize(
    "is_cpu_or_gpu",
    [
        pytest.param("cpu", marks=pytest.mark.cpu_run),
        pytest.param("gpu", marks=pytest.mark.gpu_run),
    ]
)
def test_status_before_initialize(is_cpu_or_gpu, cleanup_after_initialize) -> None:
    """
    Integration test to verify 'dss status' command before initialization.
    """

    # Run the status command
    result = subprocess.run(
        ["dss", "status", "--kubeconfig", KUBECONFIG], capture_output=True, text=True
    )

    # Check if the command executed successfully
    assert result.returncode == 0

    # Check if the output indicates MLflow deployment is not ready
    assert "MLflow deployment: Not ready" in result.stdout

    if is_cpu_or_gpu == "cpu":
        # Check if the output indicates GPU acceleration is disabled
        assert "GPU acceleration: Disabled" in result.stdout
    elif is_cpu_or_gpu == "gpu":
        # Check if the output indicates GPU acceleration is enabled and captures a GPU model
        assert "GPU acceleration: Enabled" in result.stdout
        assert (
                "NVIDIA" in result.stdout
        )  # This checks for the presence of NVIDIA in the log, assuming only NVIDIA GPUs are used.
    else:
        raise ValueError(f"Invalid value for is_cpu_or_gpu: {is_cpu_or_gpu}")
```

Up to you if you want to use it or leave the tests as is.  I don't think there's a downside to using it (and using it in the other duplicated tests below)